### PR TITLE
feat: actsvg plugin for new detector infrastructure (cylindrical)

### PR DESCRIPTION
**Pull request by asalzburger**
This PR includes the necessary `ActSVG` converters for the for `Portal`, `DetectorVolume` and `Detector`. It currently is pretty much restricted to a cylindrical setup, but can be easily augmented to also include a cubic detector setup.

`Portal` display includes (optionally coloured) arrows that point to the volume in question:
 
Screenshot 2023-06-21 at 13 22 16

`DetectorVolume` display includes contained surfaces (and grid structure if given(

Screenshot 2023-06-21 at 13 21 47

`Detector` display puts this all together:

Screenshot 2023-06-21 at 13 21 32

### CODE REVIEW

File: Core/src/Detector/DetectorVolumeBuilder.cpp [hunk 1]
```diff
@@ -74,8 +74,10 @@ Acts::Experimental::DetectorVolumeBuilder::construct(
     // Construct the DetectorVolume
     dVolume = DetectorVolumeFactory::construct(
         portalGenerator, gctx, m_cfg.name, transform, std::move(bounds),
-        surfaces, volumes, std::move(surfacesUpdator),
-        std::move(volumeUpdator));
+        surfaces, volumes, std::move(volumeUpdator),
```
Comment (asalzburger) [inline Core/src/Detector/DetectorVolumeBuilder.cpp:77]: That was a BUG! wrong ordering.

**Comment by paulgessinger** [comment]
Looks 🔥

File: Plugins/ActSVG/src/DetectorVolumeSvgConverter.cpp [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+std::tuple<Acts::Svg::ProtoVolume, Acts::Svg::ProtoIndexedSurfaceGrid>
+Acts::Svg::DetectorVolumeConverter::convert(
+    const GeometryContext& gctx, const Experimental::DetectorVolume& dVolume,
+    const DetectorVolumeConverter::Options& volumeOptions) {
+  // The local logger
+  ACTS_LOCAL_LOGGER(
+      getDefaultLogger("DetectorVolumeSvgConverter", volumeOptions.logLevel));
+
+  ProtoVolume pVolume;
+  pVolume._name = dVolume.name();
```
Comment (andiwand) [inline Plugins/ActSVG/src/DetectorVolumeSvgConverter.cpp:31]: what is the reason for `_`? is this something the user should not rely on?

File: Plugins/ActSVG/src/PortalSvgConverter.cpp [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+        dynamic_cast<const Experimental::BoundVolumesGrid1Impl*>(instance);
+    if (multiLink != nullptr) {
+      auto pLinks = convertMultiLink(gctx, *multiLink, surface, rPos,
+                                     portalOptions, sign);
+
+      pPortal._volume_links.insert(pPortal._volume_links.end(), pLinks.begin(),
+                                   pLinks.end());
+    }
+    // Switch to the other side
+    sign += 2;
```
Comment (andiwand) [inline Plugins/ActSVG/src/PortalSvgConverter.cpp:155]: should we somehow guarantee that this can run only a single time?

**Review by andiwand — APPROVED** [review]
left a few comments but looks already pretty good

this will conflict with https://github.com/acts-project/acts/pull/2214 so I need to catch up when this is merged

File: Plugins/ActSVG/src/SurfaceSvgConverter.cpp [hunk 4]
```diff
@@ -111,7 +111,9 @@ Acts::Svg::ProtoSurface Acts::Svg::SurfaceConverter::convert(
     // Set the openings
     actsvg::scalar ri = static_cast<actsvg::scalar>(boundValues[0]);
     actsvg::scalar ro = static_cast<actsvg::scalar>(boundValues[1]);
+    actsvg::scalar zp = static_cast<actsvg::scalar>(surface.center(gctx).z());
     pSurface._radii = {ri, ro};
+    pSurface._zparameters = {zp, zp};
```
[the remark this answers is not in the document]
Comment (asalzburger) [reply Plugins/ActSVG/src/SurfaceSvgConverter.cpp:116]: Indeed.

File: Core/src/Detector/DetectorVolumeBuilder.cpp (hunk 1 above)
Comment (asalzburger) [reply Core/src/Detector/DetectorVolumeBuilder.cpp:77]: Thanks, that will be needed.

File: Plugins/ActSVG/src/DetectorSvgConverter.cpp [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+#include "Acts/Utilities/Logger.hpp"
+
+#include <map>
+
+Acts::Svg::ProtoDetector Acts::Svg::DetectorConverter::convert(
+    const GeometryContext& gctx, const Experimental::Detector& detector,
+    const DetectorConverter::Options& detectorOptions) {
+  // The local logger
+  ACTS_LOCAL_LOGGER(
+      getDefaultLogger("DetectorVolumeConverter", detectorOptions.logLevel));
```
[the remark this answers is not in the document]
Comment (asalzburger) [reply Plugins/ActSVG/src/DetectorSvgConverter.cpp:26]: I removed the logging entirely - this is a converter, most of the time the logger wasn't even used, and sometimes it just said: "converting X".

File: Plugins/ActSVG/src/DetectorVolumeSvgConverter.cpp [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+
+#include "Acts/Detector/DetectorVolume.hpp"
+#include "Acts/Detector/Portal.hpp"
+#include "Acts/Geometry/GeometryIdentifier.hpp"
+#include "Acts/Geometry/VolumeBounds.hpp"
+#include "Acts/Plugins/ActSVG/IndexedSurfacesSvgConverter.hpp"
+#include "Acts/Utilities/Enumerate.hpp"
+#include "Acts/Utilities/Logger.hpp"
+
+#include <set>
```
[the remark this answers is not in the document]
Comment (asalzburger) [reply Plugins/ActSVG/src/DetectorVolumeSvgConverter.cpp:19]: Removed.

File: Plugins/ActSVG/src/PortalSvgConverter.cpp (hunk 3 above)
Comment (asalzburger) [reply Plugins/ActSVG/src/PortalSvgConverter.cpp:155]: At the moment this is safe, there is an array of size 2 over which it runs, I would like to address the storage and addressing of the `Portal::Forward`, `Portal::Backward` attachment though and not leave it in an implicit `std::array<X,2u>` but that would be in another PR unrelated to SVG.

**Review by andiwand — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results
### :white_check_mark: All tests successful

|status|job|
|------|---|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31088103) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31088102) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31088101) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31088100) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsTrkSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31088099) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31088098) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31088097) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31088096) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31088095) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31088094) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31088093) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31088092) |
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31088091) |
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31088090) |
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31088089) |
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31088088) |
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31088087) |

**Pull request merged**